packages/openapi2-parser/test/schema-test.js

```diff
@@ -720,7 +720,90 @@ describe('JSON Schema to Data Structure', () => {
       expect(dataStructure.content).to.be.instanceof(ObjectElement);
 
       expect(dataStructure.content.length).to.equal(0);
-      expect(dataStructure.content.attributes.getValue('typeAttributes')).to.deep.equal(['fixed-type']);
+      expect(dataStructure.content.attributes.getValue('typeAttributes')).to.deep.equal(['fixedType']);
+    });
+
+    it('produces object with additionalProperties as false with required properties', () => {
+      const schema = {
+        type: 'object',
+        required: ['type'],
+        additionalProperties: false,
+      };
+
+      const dataStructure = schemaToDataStructure(schema);
+
+      expect(dataStructure.element).to.equal('dataStructure');
+      expect(dataStructure.content).to.be.instanceof(ObjectElement);
+
+      expect(dataStructure.content.length).to.equal(1);
+      expect(dataStructure.content.attributes.getValue('typeAttributes')).to.deep.equal(['fixedType']);
+
+      const type = dataStructure.content.getMember('type');
+      expect(type).to.be.instanceof(MemberElement);
+      expect(type.attributes.getValue('variable')).to.be.undefined;
+      expect(type.attributes.getValue('typeAttributes')).to.deep.equal(['required']);
+      expect(type.value).to.be.undefined;
+    });
+
+    it('produces object with additionalProperties containing a structure', () => {
+      const schema = {
+        type: 'object',
+        additionalProperties: {
+          type: 'string',
+        },
+      };
+
+      const dataStructure = schemaToDataStructure(schema);
+
+      expect(dataStructure.element).to.equal('dataStructure');
+      expect(dataStructure.content).to.be.instanceof(ObjectElement);
+
+      expect(dataStructure.content.length).to.equal(1);
+
+      const member = dataStructure.content.getMember('');
```

Comment:
Typically, the key of the member is a "sample" key, but we don't really have one so I have used an empty string.

Note this structure is equivalent to API Blueprint, except the key name where API Blueprint provides a sample `name`.

```apib
+ Response 200 (application/json)
    + Attributes (fixed-type)
        + *name* (string)
```